lib/kinesis_client/app_state/adapter.ex

```diff
@@ -0,0 +1,69 @@
+defmodule KinesisClient.Stream.AppState.Adapter do
+  @moduledoc """
+  This interface specifies what the AppState store needs to support.
+  """
+
+  alias KinesisClient.Stream.AppState.ShardLease
+
+  @doc """
+  Implement to setup any backend storage. Should not clear data as this will be called everytime a
+  `KinesisClient.Stream.Coordinator` process is started.
+  """
+  @callback initialize(app_name :: String.t(), opts :: Keyword.t()) :: :ok | {:error, any}
```

Comment:
Not that it would matter much, but could any of these be in [@optional_callbacks](https://hexdocs.pm/elixir/typespecs.html#optional-callbacks)?